cmake/scripts/generate_config.cmake

```diff
@@ -0,0 +1,35 @@
+##===----------------------------------------------------------------------===##
+#
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This file incorporates work covered by the following copyright and permission
+# notice:
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+#
+##===----------------------------------------------------------------------===##
+
+if (NOT OUTPUT_DIR)
+    set(OUTPUT_DIR "output")
+endif()
+
+set(ONEDPL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../..")
+
+file(READ ${ONEDPL_ROOT}/include/oneapi/dpl/pstl/onedpl_config.h _onedpl_version_info LIMIT 1024)
+string(REGEX REPLACE ".*#define ONEDPL_VERSION_MAJOR ([0-9]+).*" "\\1" _onedpl_ver_major "${_onedpl_version_info}")
+string(REGEX REPLACE ".*#define ONEDPL_VERSION_MINOR ([0-9]+).*" "\\1" _onedpl_ver_minor "${_onedpl_version_info}")
+string(REGEX REPLACE ".*#define ONEDPL_VERSION_PATCH ([0-9]+).*" "\\1" _onedpl_ver_patch "${_onedpl_version_info}")
+
+set(PROJECT_VERSION "${_onedpl_ver_major}.${_onedpl_ver_minor}.${_onedpl_ver_patch}")
+
+configure_file("${ONEDPL_ROOT}/cmake/templates/oneDPLConfig.cmake.in"
+               "${OUTPUT_DIR}/oneDPLConfig.cmake"
+               COPYONLY)
+configure_file("${ONEDPL_ROOT}/cmake/templates/oneDPLConfigVersion.cmake.in"
+               "${OUTPUT_DIR}/oneDPLConfigVersion.cmake"
+               @ONLY)
```

Comment:
What does this parameter mean? And why do we use `COPYONLY` for oneDPLComfig.cmake.in?

Reply:
From https://cmake.org/cmake/help/latest/command/configure_file.html:

> COPYONLY
> Copy the file without replacing any variable references or other content. This option may not be used with NEWLINE_STYLE.

I use it for oneDPLConfig.cmake.in because there is no need to change this file, just copy.

> @ONLY
> Restrict variable replacement to references of the form @VAR@. This is useful for configuring scripts that use ${VAR} syntax.

I use it for oneDPLConfigVersion.cmake.in because I need to replace `@PROJECT_VERSION@` with parsed project version.